chia/full_node/coin_store.py

```diff
@@ -192,6 +192,33 @@ async def get_coin_records_by_puzzle_hashes(
             coins.add(CoinRecord(coin, row[1], row[2], row[3], row[4], row[8]))
         return list(coins)
 
+    async def get_coin_records_by_names(
+        self,
+        include_spent_coins: bool,
+        names: List[bytes32],
+        start_height: uint32 = uint32(0),
+        end_height: uint32 = uint32((2 ** 32) - 1),
+    ) -> List[CoinRecord]:
+        if len(names) == 0:
+            return []
+
+        coins = set()
+        names_db = tuple([name.hex() for name in names])
+        cursor = await self.coin_record_db.execute(
+            f'SELECT * from coin_record WHERE coin_name in ({"?," * (len(names_db) - 1)}?) '
```

Comment:
This will probably perform poorly due to spent:
Check this PR: https://github.com/Chia-Network/chia-blockchain/pull/7794/files

Reply:
If I try to do INDEXED_BY coin_name, it says there is no such index.  If I index it by puzzle hash, it is actually slower than it currently stands.  coin_name is the PRIMARY_KEY so I think this is as optimized as it gets?